src/asm/lexer.c

```diff
@@ -1174,6 +1174,95 @@ static int readIdentifier(char firstChar)
 	return tokenType;
 }
 
+/* Function to read identifiers & keywords for macros within macros */
+
+static bool startsMacroWithinMacro(int c)
```

Comment:
There must be a similar check elsewhere, and should be merged with this. Preferably, the factored-out function should be placed close to the function that actually performs identifier lexing, so that the character sets are kept in sync during future modifications.

Reply:
`startsMacroWithinMacro` was based on `startsIdentifier`. They both allow `A-Za-z_`, but `startsIdentifier` also allows `.` for local labels, whereas `startsMacroWithinMacro` also allows `\\` for macro arguments or line continuations.

Reply:
I'd suggest having a `startsIdentifier` and `startsScopedIdentifier`, perhaps? The latter would accept `.`, but not the former.